app/js/ui.js

```diff
@@ -38,10 +44,60 @@ class Controller {
   }
 
   initTemplateBlocks() {
-    const blocks = document.querySelectorAll('.block.template');
-    for (let i = 0; i < blocks.length; i++) {
-      const block = blocks[i];
-      this.templateBlocks.push(new TemplateBlock(block));
+    const templateBlocks = document.querySelectorAll('.block.template');
+    for (let i = 0; i < templateBlocks.length; i++) {
+      const block = templateBlocks[i];
+      this.templateBlocks.push(new blocks.TemplateBlock(block));
+    }
+  }
+
+  createBlock(blockType, x, y) {
```

Comment:
I tried to make it as easy as possible to add blocks when loading them from a file.